sycl/source/detail/graph_impl.hpp

```diff
@@ -191,6 +181,25 @@ struct graph_impl {
     MEventsMap[EventImpl] = NodeImpl;
   }
 
+  std::shared_ptr<sycl::detail::event_impl>
+  get_event_for_node(std::shared_ptr<node_impl> NodeImpl) const {
+    if (auto EventImpl = std::find_if(
+            MEventsMap.begin(), MEventsMap.end(),
+            [NodeImpl](auto &it) { return it.second == NodeImpl; });
+        EventImpl != MEventsMap.end()) {
+      return EventImpl->first;
+    }
+
+    throw sycl::exception(
+        sycl::errc::invalid,
+        "No event has been recorded for the specified graph node");
+  }
+
+  /// Adds subgraph nodes from an executable graph to this graph, returns
```

Comment:
I guess this doesn't purely relate to your PR, but it looks like we are inconsistent with our commenting format (`//` vs. `///`).